### PR TITLE
Looping on a list rather than iterator to avoid RuntimeError

### PR DESCRIPTION
**Pull request by bsipocz**
This was recently changed in #5429, but using the iterator instead of a static list raises a RuntimeError e.g. here: https://travis-ci.org/astropy/saba/jobs/179464471

```
copying saba/tests/coveragerc -> build/lib.macosx-10.10-x86_64-3.4/saba/tests
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/saba-test-921eh7vo/lib.macosx-10.10-x86_64-3.4/saba/__init__.py", line 15, in <module>
    from .main import SherpaFitter, SherpaMCMC, Stat, OptMethod, EstMethod, Dataset, ConvertedModel
  File "/private/var/folders/dc/hsm7tqpx2d57n7vb3k1l81xw0000gq/T/saba-test-921eh7vo/lib.macosx-10.10-x86_64-3.4/saba/main.py", line 21, in <module>
    with catch_warnings(AstropyUserWarning) as warns:
  File "/Users/bsipocz/munka/devel/astropy/astropy/tests/helper.py", line 309, in __enter__
    treat_deprecations_as_exceptions()
  File "/Users/bsipocz/munka/devel/astropy/astropy/tests/helper.py", line 214, in treat_deprecations_as_exceptions
    for module in six.itervalues(sys.modules):
RuntimeError: dictionary changed size during iteration
```

### CODE REVIEW

**Comment by pllim** [comment]
@MSeifert04  ?

**Comment by MSeifert04** [comment]
I've really no idea why the `sys.modules` should change when deleting their `__warningregistry__`.

@bsipocz Do you know what caused this and why Travis hasn't picked it up (regression test?)?

File: astropy/tests/helper.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def treat_deprecations_as_exceptions():
     warning state.
     """
     # First, totally reset the warning state
-    for module in six.itervalues(sys.modules):
+    for module in list(six.itervalues(sys.modules)):
```
Comment (MSeifert04) [inline astropy/tests/helper.py]: Could you add a small comment before, something like `The modules may change during the iteration - copy the current state to a list. [#5513]`

**Comment by bsipocz** [comment]
Actually I couldn't spot why it comes up at al based on what saba does (their ways seems sometimes a bit hacky, but nevertheless totally valid).

Anyway with a bit of debug printing, the loop adds one extra member to ``sys.modules``  whet it gets to the ``py.error`` aliasmodule (alias to py._error.error), so brings in py._error to the modules. I don't have any clue why we didn't see this anywhere else, neither for the core or the affiliates.

**Comment by bsipocz** [comment]
Weird, I was sure circle ci should understand ``[skip ci]``, apparently it doesn't.

**Comment by MSeifert04** [comment]
The docs say it should know both but in my experience `[ci skip]` works more often than `[skip ci]`. (could also be the other way around).

Thanks for the explanation. 👍

**Pull request merged**